core/unit_test/tools/TestScopedProfileRegion.cpp

```diff
@@ -0,0 +1,83 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Profiling_ScopedProfileRegion.hpp>
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <stack>
+
+namespace {
+std::stack<std::string> test_region_stack;
+
+// NOTE: cannot use lambdas because they can only be converted to function
+// pointers if they don't capture anything
+void test_push_region(char const *label) {
+  ASSERT_TRUE(true) << std::string("push ") + label;
```

Comment:
I don't think it actually works.  I can't get the message to get printed.

Reply:
Yeah, me neither.

Reply:
Yes.  Or if we want to see the info just use `printf("[          ] %s\n", ...)`